# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/michiel/ruby/openssl.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./ext/digest/md5/md5ossl.c.orig	2005-07-18 16:11:09.000000000 +0200
+++ ./ext/digest/md5/md5ossl.c	2005-07-18 16:23:18.000000000 +0200
@@ -1,5 +1,7 @@
 /* $Id: md5ossl.c,v 1.2 2003/01/06 11:47:53 knu Exp $ */
 
+#include <sys/types.h>
+
 #include "md5ossl.h"
 #include <sys/types.h>
 #include <stdio.h>
--- ./ext/digest/sha1/sha1ossl.c.orig	2005-07-18 16:11:45.000000000 +0200
+++ ./ext/digest/sha1/sha1ossl.c	2005-07-18 16:23:32.000000000 +0200
@@ -1,5 +1,7 @@
 /* $Id: sha1ossl.c,v 1.1 2002/09/26 17:44:33 knu Exp $ */
 
+#include <sys/types.h>
+
 #include "sha1ossl.h"
 #include "defs.h"
 #include <assert.h>
--- ./ext/digest/rmd160/rmd160ossl.c.orig	2005-07-18 16:12:11.000000000 +0200
+++ ./ext/digest/rmd160/rmd160ossl.c	2005-07-18 16:23:26.000000000 +0200
@@ -1,5 +1,7 @@
 /* $Id: rmd160ossl.c,v 1.1 2002/09/26 17:26:46 knu Exp $ */
 
+#include <sys/types.h>
+
 #include "rmd160ossl.h"
 #include "defs.h"
 #include <assert.h>
--- ./ext/openssl/ossl_x509store.c.orig	2005-07-18 16:13:49.000000000 +0200
+++ ./ext/openssl/ossl_x509store.c	2005-07-18 16:22:03.000000000 +0200
@@ -538,7 +538,7 @@
 
     if(NIL_P(time)) {
 	GetX509StCtx(self, store);
-	store->flags &= ~X509_V_FLAG_USE_CHECK_TIME;
+	store->param->flags &= ~X509_V_FLAG_USE_CHECK_TIME;
     }
     else {
 	long t = NUM2LONG(rb_Integer(time));
